http_funcs/api_handlers.go

```diff
@@ -90,18 +90,14 @@ func (h *HttpServ) AddMW(w http.ResponseWriter, r *http.Request) {
 
 	log.WithField("path", tmpFileLoc).Info("saved bundle")
 
-	defer func() {
-		if errRemove := os.Remove(tmpFileLoc); errRemove != nil && !os.IsNotExist(errRemove) {
-			log.WithError(errRemove).WithField("temp-file", tmpFileLoc).Warning("could not remove temp file")
-		}
-	}()
-
-	// By default, assume this is a plugin bundle
+	// By default, assume this is a new plugin bundle, which would require a process restart
 	processor := h.api.HandleNewBundle
+	restartNeeded := true
 
 	if r.FormValue("store_only") == "true" {
```

Comment:
Good point. There would be implications on the client side as well, though.

I was thinking about swapping these around anyway, so that `overseer.Restart()` could be deferred instead of adding that `bool`.